curator-client/src/main/java/org/apache/curator/ConnectionState.java

```diff
@@ -199,13 +202,28 @@ private synchronized void checkTimeouts() throws Exception
                     {
                         log.error(String.format("Connection timed out for connection string (%s) and timeout (%d) / elapsed (%d)", zooKeeper.getConnectionString(), connectionTimeoutMs, elapsed), connectionLossException);
                     }
-                    tracer.get().addCount("connections-timed-out", 1);
+                    new EventTrace("connections-timed-out", tracer.get(), getSessionId()).commit();
                     throw connectionLossException;
                 }
             }
         }
     }
 
+    /**
+     * Return the current session id
+     */
+    public long getSessionId() {
+        long sessionId = -1;
+        if (isConnected()) {
+            try {
+                sessionId = getZooKeeper().getSessionId();
```

Comment:
Session id is a useful information to track the client side issue, it's kind of identity, so we would like to include it. I'll change it to use zookeeper.getZookeeper() directly here.